site/cds_rdm/inspire_harvester/transform_entry.py

```diff
@@ -127,12 +135,39 @@ def _transform_titles(self):
 
         return rdm_title, rdm_additional_titles
 
+    def _validate_imprint(self):
+        """Validate that record has only 1 imprint."""
+        imprints = self.inspire_metadata.get("imprints", [])
+
+        if not imprints:
+            return {}
+        if len(imprints) > 1:
+            self.metadata_errors.append(
+                f"More than 1 imprint found. INSPIRE record id: {self.inspire_metadata.get('control_number')}."
+            )
+            return {}
+
+        return imprints[0]
+
+    def _transform_publisher(self):
+        """Mapping of publisher."""
+        imprint = self._validate_imprint()
+        return imprint.get("publisher")
+
     def _transform_publication_date(self):
         """Mapping of INSPIRE thesis_info.date to metadata.publication_date."""
-        thesis_date = self.inspire_metadata.get("thesis_info", {}).get(
-            "date"
-        ) or self.inspire_metadata.get("thesis_info", {}).get("defense_date")
+        imprint = self._validate_imprint()
```

Comment:
this can be also addressed in a new PR. I think we can split validation from transformation, and not to run validation functions twice.